pkg/csi/service/vanilla/controller.go

```diff
@@ -1349,10 +1351,208 @@ func (c *controller) ValidateVolumeCapabilities(ctx context.Context, req *csi.Va
+			endIndex = len(queryResult.Volumes)
+		}
+		log.Debug("Starting token: ", startingToken, " End index: ", endIndex, "Length of Query volume result ",
+			len(queryResult.Volumes), "Max entries: ", maxEntries)
+		entries, nextToken, volumeType, err = c.processQueryResultsListVolumes(ctx, startingToken, endIndex,
+			queryResult, allNodeVMs)
+		if err != nil {
+			return nil, csifault.CSIInternalFault, fmt.Errorf("Error while processing query results for list " +
+				" volumes, err: %v", err)
+		}
+		resp := &csi.ListVolumesResponse{
+			Entries:   entries,
+			NextToken: nextToken,
+		}
+
+		log.Infof("ListVolumes served %d results, token for next set: %s", len(entries), nextToken)
+		return resp, "", nil
+	}
+	listVolResponse, faultType, err := listVolumesInternal()
+	log.Debug("List volume response: ", listVolResponse)
+	if err != nil {
+		prometheus.CsiControlOpsHistVec.WithLabelValues(volumeType, prometheus.PrometheusListVolumesOpType,
+			prometheus.PrometheusFailStatus, faultType).Observe(time.Since(start).Seconds())
+	} else {
+		prometheus.CsiControlOpsHistVec.WithLabelValues(volumeType, prometheus.PrometheusListVolumesOpType,
+			prometheus.PrometheusPassStatus, faultType).Observe(time.Since(start).Seconds())
+	}
+	return listVolResponse, err
+}
+
+
+func (c *controller) processQueryResultsListVolumes(ctx context.Context, startingToken int, endIndex int,
+	queryResult *cnstypes.CnsQueryResult, allNodeVMs []*cnsvsphere.VirtualMachine) ([]*csi.ListVolumesResponse_Entry,
+	string, string, error){
+
+	volumeType := ""
+	nextToken := ""
+	log := logger.GetLogger(ctx)
+	var entries []*csi.ListVolumesResponse_Entry
+
+	for i := startingToken; i < endIndex; i++ {
+		if queryResult.Volumes[i].VolumeType == common.FileVolumeType {
+			volumeType = prometheus.PrometheusFileVolumeType
+			fileVolID := queryResult.Volumes[i].VolumeId.Id
+			log.Info("File Volume ID: ", fileVolID)
+
+			// Populate csi.Volume info for the given volume
+			fileVolumeInfo := &csi.Volume{
+				VolumeId:      fileVolID,
+			}
+
+			// Getting published nodes
+			publishedNodeIds := commonco.ContainerOrchestratorUtility.GetNodesForVolumes(ctx, []string{fileVolID})
+			nodeVMObj, err := c.nodeMgr.GetNodeByName(ctx, publishedNodeIds[fileVolID][0])
+			if err != nil {
+				log.Errorf("Failed to get node vm object from the node name, err:%v", err)
+				return entries, nextToken, volumeType, err
+			}
+			nodeVMUUID := nodeVMObj.UUID
+			log.Info("UUID of the node on which file volume is mounted: ", nodeVMUUID)
+
+			volStatus := &csi.ListVolumesResponse_VolumeStatus{
+				PublishedNodeIds: []string{nodeVMUUID},
+			}
+
+			// Populate List Volumes Entry Response
+			entry := &csi.ListVolumesResponse_Entry{
+				Volume: fileVolumeInfo,
+				Status: volStatus,
+			}
+
+			entries = append(entries, entry)
+		} else {
+			volumeType = prometheus.PrometheusBlockVolumeType
+			blockVolID := queryResult.Volumes[i].VolumeId.Id
+			log.Info("Block Volume ID: ", blockVolID)
```

Comment:
Don't use info log. Use debug. Also can use please consistently use debugf or logf instead of just info.